src/app/InteractionModelUtilities.h

```diff
@@ -0,0 +1,115 @@
+    {
+        return ((aCommandPathParams.mClusterId == mClusterId) && (aCommandPathParams.mCommandId == mCommandId) &&
+                (mFlags.Has(aCommandPathParams.mFlags)) &&
+                (mFlags.Has(CommandPathFlags::kEndpointIdValid) ? aCommandPathParams.mEndpointId == mEndpointId : false) &&
+                (mFlags.Has(CommandPathFlags::kGroupIdValid) ? aCommandPathParams.mGroupId == mGroupId : false));
+    }
+    EndpointId mEndpointId            = 0;
+    GroupId mGroupId                  = 0;
+    ClusterId mClusterId              = 0;
+    CommandId mCommandId              = 0;
+    BitFlags<CommandPathFlags> mFlags = CommandPathFlags::kEndpointIdValid;
+};
+
+enum class AttributePathFlags : uint8_t
+{
+    kFieldIdValid   = 0x01,
+    kListIndexValid = 0x02,
+};
+
+struct AttributePathParams
+{
+    AttributePathParams(chip::NodeId aNodeId, chip::EndpointId aEndpointId, chip::ClusterId aClusterId, FieldId aFieldId,
+                        ListIndex aListIndex, const BitFlags<AttributePathFlags> & aFlags) :
+        mNodeId(aNodeId),
+        mEndpointId(aEndpointId), mClusterId(aClusterId), mFieldId(aFieldId), mListIndex(aListIndex), mFlags(aFlags)
+    {}
+
+    bool operator==(const AttributePathParams & aAttributePathParams) const
+    {
+        return ((aAttributePathParams.mNodeId == mNodeId) && (aAttributePathParams.mEndpointId == mEndpointId) &&
+                (aAttributePathParams.mClusterId == mClusterId) && (mFlags.Has(aAttributePathParams.mFlags)) &&
+                (mFlags.Has(AttributePathFlags::kFieldIdValid) ? aAttributePathParams.mFieldId == mFieldId : false) &&
+                (mFlags.Has(AttributePathFlags::kListIndexValid) ? aAttributePathParams.mListIndex == mListIndex : false));
+    }
+    chip::NodeId mNodeId                = 0;
+    chip::EndpointId mEndpointId        = 0;
+    chip::ClusterId mClusterId          = 0;
+    chip::FieldId mFieldId              = 0;
+    chip::ListIndex mListIndex          = 0;
+    BitFlags<AttributePathFlags> mFlags = AttributePathFlags::kFieldIdValid;
+};
+
+struct EventPathParams
+{
+    EventPathParams(chip::NodeId aNodeId, chip::EndpointId aEndpointId, chip::ClusterId aClusterId, chip::EventId aEventId,
+                    bool aIsUrgent) :
+        mNodeId(aNodeId),
+        mEndpointId(aEndpointId), mClusterId(aClusterId), mEventId(aEventId), mIsUrgent(aIsUrgent)
+    {}
+    bool operator==(const EventPathParams & aEventPathParams) const
+    {
+        return ((aEventPathParams.mNodeId == mNodeId) && (aEventPathParams.mEndpointId == mEndpointId) &&
+                (aEventPathParams.mClusterId == mClusterId) && (aEventPathParams.mEventId == mEventId));
+    }
+    NodeId mNodeId         = 0;
+    EndpointId mEndpointId = 0;
+    ClusterId mClusterId   = 0;
+    EventId mEventId       = 0;
+    bool mIsUrgent         = false;
+};
+
+struct ClusterInfo
+{
+    bool IsDirty(void) { return mDirty; }
```

Comment:
nit: remove `(void)` as argument - CHIP style is to not have this pattern (mainly because there are clang cleanups that can remove it but not the other way around)

Reply:
Done, thanks